src/browser/helper.ts

```diff
@@ -28,3 +28,67 @@ export async function findApi(axios: AxiosInstance, apiName: string, version?: s
         return apiGroup.name === apiName;
     }).length > 0;
 }
+
+
+export interface IRequestConfig extends AxiosRequestConfig {
+}
+
+export interface IResponse<T> extends AxiosResponse<T> {
+    data: T;
+    status: number;
+    statusText: string;
+    headers: any;
+    config: IRequestConfig;
+    request?: any;
+}
+
+export interface IRequestError extends Error {
+    status?: number;
+    config: AxiosRequestConfig;
+    request?: any;
+    response?: IResponse<any>;
+}
+
+export class RequestError implements IRequestError {
+
+    status: number | undefined;
+    name: string;
+    message: string;
+    config: AxiosRequestConfig;
+    request: any;
+    response: AxiosResponse | undefined;
+
+    constructor(error: AxiosError) {
+        if (error.code) {
+            this.status = Number(error.code);
+        }
+        this.name = error.name;
+        this.config = error.config;
+        if (error.request) {
+            this.request = error.request;
+        }
+        if (error.response) {
+            this.response = error.response;
+        }
+        if ((this.status === -1 || !this.status) && (!this.response  || (this.response && !this.response.status))) {
+            // request is interrupted, there is not even an error
+            this.message = `network issues occured while requesting "${this.config.url}".`;
+        } else if (this.response && this.response.data && this.response.data.message) {
+            // che Server error that should be self-descriptive
+            this.message = this.response.data.message;
+        } else {
+            // the error is not from Che Server, so error may be in html format that we're not able to handle.
```

Comment:
It's not returned by Che Server, right?

Do we have the same for node version(this is for browser only I assume)?

Reply:
These changes are for browser side only for now. I _thought_ that this might be hit if we sent an expired token and then we'd be served with the unauthorized page